### PR TITLE
#1497 The feature to compress pictures

### PR DESCRIPTION
**Pull request by iamjaspreetsingh**
For issue #1497, one part is completed.   

Changes: Created a new activity and few dialog boxes for the feature to compress images to smaller no. of pixels or size (both the options are given separately). The icon of Compress is given below each picture alongside Edit icon.  

Screenshots for the change: 
screenshot_2018-01-02-00-25-57-306_org fossasia phimpme
screenshot_2018-01-02-00-26-02-049_org fossasia phimpme
screenshot_2018-01-02-00-26-13-534_org fossasia phimpme
screenshot_2018-01-02-00-26-57-554_org fossasia phimpme

@anantprsd5 @mohitmanuja 
Kindly review it please. 
Thank you

### CODE REVIEW

File: app/src/main/java/org/fossasia/phimpme/editor/CompressImageActivity.java [hunk 1]
```diff
@@ -0,0 +1,205 @@
...
+              percent.setText(progress1+"%");
+              percentagecompress=progress1;}
+
+             @Override
+             public void onStartTrackingTouch(SeekBar seekBar) {
+
+             }
+
+             @Override
+             public void onStopTrackingTouch(SeekBar seekBar) {
```
[the remark this answers is not in the document]
Comment (iamjaspreetsingh) [reply app/src/main/java/org/fossasia/phimpme/editor/CompressImageActivity.java]: These methods need to be implemented by OnSeekBarChangeListener().

File: app/src/main/java/org/fossasia/phimpme/editor/CompressImageActivity.java [hunk 2]
```diff
@@ -0,0 +1,205 @@
...
+         percentsize.setOnSeekBarChangeListener(new SeekBar.OnSeekBarChangeListener() {
+             @Override
+             public void onProgressChanged(SeekBar seekBar, int progress, boolean fromUser) {
+             int progress1=95-progress;
+              progress1=progress1-progress1%5;
+              percent.setText(progress1+"%");
+              percentagecompress=progress1;}
+
+             @Override
+             public void onStartTrackingTouch(SeekBar seekBar) {
```
[the remark this answers is not in the document]
Comment (iamjaspreetsingh) [reply app/src/main/java/org/fossasia/phimpme/editor/CompressImageActivity.java]: These methods need to be implemented by OnSeekBarChangeListener()

File: app/src/main/java/org/fossasia/phimpme/editor/CompressImageActivity.java [hunk 3]
```diff
@@ -0,0 +1,206 @@
...
+    public String saveFilePath;
+    public static final String EXTRA_OUTPUT = "extra_output";
+    public int percentagecompress=0;
+
+    @Override
+    protected void onCreate(Bundle savedInstanceState) {
+        super.onCreate(savedInstanceState);
+        setContentView(R.layout.activity_compress_image);
+        initView();
+        Button size=(Button)findViewById(R.id.size);
```
Comment (anantprsd5) [inline app/src/main/java/org/fossasia/phimpme/editor/CompressImageActivity.java]: Avoid findViewById calls. Use Butterknife instead

**Comment by iamjaspreetsingh** [comment]
Working on it..

**Comment by iamjaspreetsingh** [comment]
Please check there are no conflicts as such.
Please help me resolve them

**Comment by iamjaspreetsingh** [comment]
Codacy quality review is not getting right due to Butterknife implemented

**Comment by iamjaspreetsingh** [comment]
Conflicts are resolved and branch is updated @isuruAb

**Comment by mariobehling** [comment]
Closing due to inactivity.

**Comment by iamjaspreetsingh** [comment]
I was waiting for review @mariobehling.
Please check

**Comment by yashLadha** [comment]
You can again sent a PR for the respective issue with the changes. @iamjaspreetsingh

**Comment by iamjaspreetsingh** [comment]
Okay I am sending the PR again @yashLadha

**Pull request closed without merging**